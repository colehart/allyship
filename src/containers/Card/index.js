import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { toggleSaved } from '../../actions';
import './Card.css';

export const Card = (props) => {
  const {
    source,
    author,
    title,
    description,
    url,
    urlToImage,
    published,
    isSaved,
    toggleSaved
  } = props

  return (
    <article className='Card'>
      <div className='c-title-group'>
        <h3 className='c-title'>{title}</h3>
        <div
          className={`c-save ${ isSaved ? 'is-saved' : '' }`}
          onClick={() => toggleSaved(title)}
        >
        </div>
      </div>
      <p className='c-byline'>{published} | {source} {`${author ? `| ${author}` : ''}`}</p>
      <img className='c-image' src={urlToImage} alt='provided illustration'/>
      <p className='c-description'>{description}</p>
      <a
        className='c-link'
        href={url}
        target='_blank'
        rel='noopener noreferrer'
      >
        Read article from source
      </a>
    </article>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  toggleSaved: (headline) => dispatch(toggleSaved(headline))
})

Card.propTypes = {
  toggleSaved: PropTypes.func
}

export default connect(null, mapDispatchToProps)(Card)