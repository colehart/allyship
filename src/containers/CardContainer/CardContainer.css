.cc-title-group {
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin: 40px 0 10px;
}

.cc-back-btn {
  text-decoration: none;
  color: var(--coral);
  bottom: 15px;
  font-weight: 900;
  font-size: 1.25rem;
  cursor: pointer;
  border: none;
  background-color: transparent;
}

.cc-back-btn:hover {
  text-decoration: none;
  color: var(--turquoise);
  bottom: 15px;
}

.cc-title {
  text-align: center;
}

.CardContainer {
  grid-row: 2 / 3;
  width: 80%;
  margin: 0 auto;
  padding-top: 20px;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}
