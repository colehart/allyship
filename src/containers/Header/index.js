import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import shuttleIcon from '../../assets/images/space-shuttle.svg';
import './Header.css';

export const Header = () => {
  return (
    <div className='Header'>
      <Link
        to='/'
        className='h-logo-group'
      >
        <img
          className='h-logo'
          src={ shuttleIcon }
          alt='A space shuttle'
        />
        <h1>Allyship</h1>
      </Link>
      <nav className='h-nav-group'>
        <NavLink
          exact to='/'
          className="h-categories btn"
        >
          Categories
        </NavLink>
        <NavLink
          exact to='/saved'
          className="h-saved btn"
        >
          Saved
          <span className='h-badge'> 0</span>
        </NavLink>
      </nav>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  caughtError: state.caughtError,
  stories: state.stories
})

export default connect(mapStateToProps)(Header)