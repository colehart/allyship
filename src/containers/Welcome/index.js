import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Loader } from '../../components/Loader';
import { fetchStories } from '../../thunks/fetchStories';
import './Welcome.css';

export const Welcome = (props) => {
  const { isLoading } = props

  return (
    <main className='Welcome'>
      { isLoading ? <Loader /> : null }
      <section
        className={ isLoading ? 'hidden' : 'w-message'}
      >
        <h2>Welcome!</h2>
        <p>
          Allyship exists to help you find your way in the galaxy of knowledge and news any true ally must navigate.
        <br />
        <br />
          Research the latest news about a specific minority identity or issue.
        {/* , or choose your own query by entering a search keyword or phrase. */}
        </p>
      </section>
      <div
        className='w-category-group'
      >
        <Link
          to='/transgender'
          exact='true'
          className='btn'
          aria-label='Click here for the latest transgender news'
        >
          Transgender
        </Link>
        <Link
          to='/immigration'
          exact='true'
          className='btn'
          aria-label='Click here for the latest immigration news'
        >
          Immigration
        </Link>
        <Link
          to='/black-lives-matter'
          exact='true'
          aria-label='Click here for the latest black lives matter news'
          className='btn'
        >
          Black Lives Matter
        </Link>
      </div>
    </main>
  )
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading
})

export const mapDispatchToProps = (dispatch) => ({
  fetchStories: (url) => dispatch(fetchStories(url))
})

Welcome.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  fetchStories: PropTypes.func.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Welcome)
