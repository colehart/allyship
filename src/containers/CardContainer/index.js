import React from 'react';
import Card from '../Card'
import './CardContainer.css';

export const CardContainer = () => {
  const cards = []

  return (
    <main className="CardContainer">
      { cards }
    </main>
  )
}
