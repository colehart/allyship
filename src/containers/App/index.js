import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { Header } from '../Header';
import { Welcome } from '../Welcome';
import { CardContainer } from '../CardContainer';
import './App.css';

export const App = (props) => {
  const { isLoading } = props

  const noMatch = ({ location }) => (
    <div className='a-404'>
      <h2>404 - Your ship has gone off course.</h2>
      <p>No routes match {location.pathname}.</p>
      <p>Please click the orange buttons in this app to navigate to your desired destination.</p>
    </div>
  )

  return (
    <div className='App'>
      <Header />
      <Switch>
        <Route
          exact path='/'
          component={ Welcome }
        />
        <Route
          exact path='/saved'
          render={() => CardContainer }
        />
        <Route
          exact path='/transgender'
          render={() => CardContainer }
        />
        <Route
          exact path='/immigration'
          render={() => CardContainer }
        />
        <Route
          exact path='/black-lives-matter'
          render={() => CardContainer }
        />
        <Route
          component={noMatch}
        />
      </Switch>
      {/* <Route
        exact path='/search'
        render={() => CardContainer }
      /> */}
    </div>
  );
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading,
})

export default connect(mapStateToProps)(App);
