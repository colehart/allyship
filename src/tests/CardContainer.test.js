import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { CardContainer } from '../containers/CardContainer';

describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CardContainer />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})