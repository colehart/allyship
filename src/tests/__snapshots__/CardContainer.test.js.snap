// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer CardContainer component matches the snapshot 1`] = `
<main>
  <div
    className="cc-title-group"
  >
    <h2
      className="cc-title"
    >
      Latest Immigration News
    </h2>
  </div>
  <div
    className="CardContainer"
  >
    <Connect(Card)
      author="CLAIRE CAIN MILLER"
      content="Transgender people are appearing more often in advertisements and mass media. But when they do, its very often in stock photos that show them standing against a blank wall, or else they are hardly seen at all. The most used stock photos are close-ups of their… [+1039 chars]"
      description="The most used photo of a transgender person from Getty Images’s stock photo library shows the transgender pride symbol — but only part of the person’s face."
      isSaved={false}
      key="https://www.nytimes.com/2018/10/25/upshot/what-being-transgender-looks-like-according-to-stock-photography.html"
      published="2018-10-25"
      source="The New York Times"
      title="What Being Transgender Looks Like, According to Stock Photography"
      url="https://www.nytimes.com/2018/10/25/upshot/what-being-transgender-looks-like-according-to-stock-photography.html"
      urlToImage="https://static01.nyt.com/images/2018/10/23/upshot/oakImage-1540334490988/oakImage-1540334490988-facebookJumbo.jpg"
    />
    <Connect(Card)
      author="Victoria Rodriguez"
      content="There are many reasons you should learn how to be an ally to trans people. First, transgender rights are human rights it's that simple. Trans people regularly encounter discrimination in everyday settings, such as workplaces, schools, and doctor's offices. Th… [+8478 chars]"
      description="There are many reasons you should learn how to be an ally to trans people. First, transgender rights are human rights — it's that simple. Trans people regularly encounter discrimination in everyday settings, such as workplaces, schools, and doctor's offices. …"
      isSaved={false}
      key="https://mashable.com/article/transgender-rights-how-to-be-an-ally/"
      published="2018-10-23"
      source="Mashable"
      title="How to be an ally who stands with transgender people"
      url="https://mashable.com/article/transgender-rights-how-to-be-an-ally/"
      urlToImage="https://i.amz.mshcdn.com/5pzXh7-L0211KSsQEfj9UV4v23o=/1200x630/2018%2F10%2F23%2Fd5%2Febde5e356f604ebeaa4548bdc4c4da6a.9e146.jpg"
    />
  </div>
</main>
`;

exports[`CardContainer CardContainer component matches the snapshot if location is /saved 1`] = `
<main>
  <div
    className="cc-title-group"
  >
    <button
      className="cc-back-btn"
      onClick={[Function]}
    >
      ← Back
    </button>
    <h2
      className="cc-title"
    >
      Saved Stories
    </h2>
    <div
      className="cc-placeholder"
    />
  </div>
  <div
    className="CardContainer"
  >
    <div
      className="cc-no-saved"
    >
      <h3>
        You have no saved stories.
      </h3>
      <p>
        Click the sun icon in the upper right corner of any article to save a story.
      </p>
    </div>
  </div>
</main>
`;

exports[`CardContainer CardContainer component should render the Loader if isLoading(true) 1`] = `
<main>
  <div
    className="cc-title-group"
  >
    <h2
      className="cc-title"
    >
      Latest Immigration News
    </h2>
  </div>
  <div
    className="CardContainer"
  >
    <Loader />
  </div>
</main>
`;
